articles/secret-key-cryptography-in-js/index.md

```diff
@@ -0,0 +1,138 @@
+
+### Implementing secret key cryptography in JavaScript.
+
+For purposes of this article, we shall use the crypto module which is a core module in `Node.js` so you won't have to install any third-party module for the project.
+
+#### Forming the Puzzle.
+
+For us to implement secret key cryptography in JavaScript, we need to form a practical puzzle. Let's assume, we have two communicating parties, David and Viktoria. For secret key cryptography, if David wants to send a message to Viktoria, he encrypts the message using the secret key to get the cipher text. The cipher text is transported over a secure medium whereas the secret key is transported over an insecure medium. On reaching the other end, Viktoria uses her secret key to decode the message and get the plain text. She shall follow the same process if she intends to send a reply.
+
+#### Defining variables and modules
+
+```javascript
+    const crypto = require('crypto');
+
+    const alg = `aes-192-cbc`;
+
+    const pass_key = `david_viktoria`;
+
+    const salt = 'salt';
+
+    const keylen = 24;
+
+    const shared_key = crypto.scryptSync(pass_key,salt,keylen);
+
+    const one_time_code = crypto.randomBytes(16);
+
+    const cipher = crypto.createCipheriv(alg,shared_key,one_time_code);
+
+    const decipher = crypto.createDecipheriv(alg,shared_key,one_time_code);
+```
+From the above, we define the module crypto which is a core module meaning. From there, we specify the algorithm we are going to use since crypto can support more than one algorithm. We shall use `AES`. We then come up with the neccessary parameters to get the shared key. A one time code is also needed in generating the cipher and the decipher. Based on the above, we have everything for encryption and decryption.
+
+#### David encrypting the message
+
+```javascript
+    function davidEncrypting(){
+    const davids_message = "Hello there";
+
+    cipher.on('readable',() => {
+        let _cipher_text = cipher.read()
+        if(_cipher_text){
+
+            cipher_text = _cipher_text.toString('hex');
+
+        }
+    });
+
+    cipher.write(davids_message);
+
+    cipher.end();
+   
+    };
+```
+
+When you are encrypting, you will need the message to be encrypted and from this instance, its coming from David. From there we need to listen to the readable event of the cipher since that is where we are going to get the cipher text. We write the message to the cipher and then close the cipher.
+
+#### Viktoria decrypting the message
+
+```javascript
+    decipher.on('readable',() => {
+
+        let _plain_text = decipher.read();
+
+        if(_plain_text){
+
+            console.log(_plain_text.toString('utf8'));
+            
+        };
+
+
+    });
+
+    decipher.write(cipher_text,'hex');
+
+    decipher.end();
+```
```

Comment:
```Javascript
function viktoriaDecrypting(){

    decipher.on('readable',() => {

        let _plain_text = decipher.read();

        if(_plain_text){

            console.log(_plain_text.toString('utf8'));
            
        };


    });

    decipher.write(cipher_text,'hex');

    decipher.end();

}
```
instead of the function here, just making sure the code is consistent with the working code.